src/components/scenes/CoinRankingDetailsScene.tsx

```diff
@@ -218,6 +271,176 @@ const CoinRankingDetailsSceneComponent = (props: Props) => {
+        fiatCurrencyCode: fiatCurrencyCode
+      })
+    }
+
+    return targetWallet
+  }
+
+  const handleBuyPress = useHandler(async () => {
+    if (edgeAsset == null) return
+    const wallet = await getWallet()
+    if (wallet == null) return
+
+    navigation.navigate('edgeTabs', {
+      screen: 'buyTab',
+      params: {
+        screen: 'pluginListBuy',
+        params: {
+          forcedWalletResult: {
+            type: 'wallet',
+            walletId: wallet.id,
+            tokenId: edgeAsset.tokenId
+          }
+        }
+      }
+    })
+  })
+
+  const handleSellPress = useHandler(async () => {
+    if (edgeAsset == null) return
+    const wallet = await getWallet()
+    if (wallet == null) return
+
+    navigation.navigate('edgeTabs', {
+      screen: 'sellTab',
+      params: {
+        screen: 'pluginListSell',
+        params: {
+          forcedWalletResult: {
+            type: 'wallet',
+            walletId: wallet.id,
+            tokenId: edgeAsset.tokenId
+          }
+        }
+      }
+    })
+  })
+
+  const handleTradePress = useHandler(async () => {
+    if (edgeAsset == null) return
+    const { tokenId } = edgeAsset
+
+    const targetWallet = await getWallet()
+    if (targetWallet == null) return
+
+    // Find the wallet with highest USD value to use as source
+    // TODO: Include tokenIds in the sort
+    const sourceWallet = Object.values(account.currencyWallets)
+      .filter(wallet => wallet.id !== targetWallet.id)
+      .sort((a, b) => {
+        const aCryptoAmount = new CryptoAmount({
+          currencyConfig: a.currencyConfig,
+          tokenId: null,
+          nativeAmount: a.balanceMap.get(null) ?? '0'
+        })
+        const aDollarValue = parseFloat(aCryptoAmount.displayDollarValue(exchangeRates, DECIMAL_PRECISION))
+
+        const bCryptoAmount = new CryptoAmount({
+          currencyConfig: b.currencyConfig,
+          tokenId: null,
+          nativeAmount: b.balanceMap.get(null) ?? '0'
+        })
+        const bDollarValue = parseFloat(bCryptoAmount.displayDollarValue(exchangeRates, DECIMAL_PRECISION))
+
+        return bDollarValue - aDollarValue
+      })[0]
+
+    // Navigate to the swap scene
+    navigation.navigate('edgeTabs', {
+      screen: 'swapTab',
+      params: {
+        screen: 'swapCreate',
+        params: {
+          fromWalletId: sourceWallet.id,
+          toWalletId: targetWallet.id,
+          toTokenId: tokenId
+        }
+      }
+    })
+  })
+
+  const isStakingAvailable = (): boolean => {
+    if (countryCode == null || edgeAsset == null) return false
+
+    // Special case for FIO because it uses it's own staking plugin
+    const isStakingSupported = SPECIAL_CURRENCY_INFO[edgeAsset.pluginId]?.isStakingSupported === true || currencyCode === 'FIO'
+    return isStakingSupported
+  }
+
+  // TODO: Broken - We need to come up with a wallet beforehand in order to get stakePlugins and stakePolicies for routing
```

Comment:
It looks like you solved this, and it's not broken anymore?

Reply:
No this is still broken, I wanted to get the review out first while I continue to tackle it